retriever/lib/engine_tools.py

```diff
@@ -273,3 +274,52 @@ def set_proxy():
                 for i in proxies:
                     os.environ[i] = os.environ[proxy]
                 break
+
+
+def xml2dict(dic, node, level):
+    """Convert xml to dict type.
+
+    """
+    vals = dict()
```

Comment:
I think this should be OrderedDict()

Reply:
@henrykironde values from val is being appended at line 307 So it might not be required